homeassistant/components/roborock/__init__.py

```diff
@@ -152,3 +165,9 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         hass.data[DOMAIN].pop(entry.entry_id)
 
     return unload_ok
+
+
+async def update_listener(hass: HomeAssistant, entry: ConfigEntry) -> None:
+    """Handle options update."""
+    # Reload entry to update data
+    await hass.config_entries.async_reload(entry.entry_id)
```

Comment:
Would it make sense to only redraw the image entities unless INLCUDE_SHARED changed?

(I don't know how much more expensive a full reload is)

Reply:
I think that adds an unneeded complexity. This gets in more cases than just the options flow change